libticalcs/trunk/tests/torture_ticalcs.c

```diff
@@ -539,6 +538,7 @@ static void torture_dbus(void)
 
 static void torture_cmdz80(void)
 {
+#ifndef __WIN32__
```

Comment:
Er... is that still necessary here and below, and if yes, why ?

Reply:
Apparently it is still needed (I'm not sure I'm understanding this correctly):
```
torture_ticalcs.c(546): warning C4003: not enough arguments for function-like macro invocation 'ti73_send_XDP'
torture_ticalcs.c(546): error C2059: syntax error: ','
```

That said, the preprocessor directive could be better targeted with `#ifndef _MSC_VER`

Reply:
Done!

Reply:
I know that when a compilation toolchain produces weird syntax errors about a valid-looking piece of code, the toolchain is not always the culprit, but... the libticalcs code being
```
static inline uint8_t tiz80_handle_to_dbus_mid_7383p(CalcHandle * handle)
{
	return (handle != NULL) ? ((handle->model == CALC_TI73) ? DBUS_MID_PC_TI73 : DBUS_MID_PC_TI83p) : 0;
}

TIEXPORT3 int TICALL tiz80_send_XDP(CalcHandle* handle, uint16_t length, uint8_t * data, uint8_t target);

#define ti73_send_XDP(handle, length, data) tiz80_send_XDP(handle, length, data, tiz80_handle_to_dbus_mid_7383p(handle))

#define PRINTF(FUNCTION, TYPE, ...) \
fprintf(stderr, "%d\t" TYPE "\n", __LINE__, FUNCTION(__VA_ARGS__))

#define INT "%d"

    PRINTF(ti73_send_XDP, INT, NULL, 0, (void *)0x12345678);
```
I can't see what kind of argument count mismatch could be at play here.

Does the error on the call to `ti73_send_XDP()` go away after rewriting it as a static inline function ?
```
static inline int ti73_send_XDP(CalcHandle *handle, uint16_t * length, uint8_t * data)
{
<TAB>return tiz80_send_XDP(handle, length, data, tiz80_handle_to_dbus_mid_7383p(handle));
}
```

Reply:
To be clear, there are many errors, not just this one, and the cause is that the macro expansion ends with a stray comma, or something like that.
Rewriting this as a function makes it work, yes.

Reply:
Boo. Talk about a chore, writing dozens of suchlike trivial inline functions.
Maybe you feel like writing a sed invocation to ease / perform the task ? ^^

Reply:
I looked at the macro-expansion-debugger thing that VS has, and at the last steps it has this:

Correct:
![image](https://github.com/debrouxl/tilibs/assets/660896/c1b6e4bf-6acc-45f2-adb5-d6d37e6f4fde)

Incorrect:
![image](https://github.com/debrouxl/tilibs/assets/660896/26317342-b84b-40f1-9d45-8c31f18a6492)

I'll see what I can do...

Reply:
I tried with ` ## __VA_ARGS__` as seen here and there, and the expansion is correct in the debugger, as well as in the code editor itself... but is not once compiler.
I guess something is broken in the compiler...

Reply:
Interestingly enough, if compiled as C++, this macro-business works as intended... (but then we have to fix all the type casts, but...)

Reply:
I have rewritten the test in C++, which allows for better typing anyway...

Reply:
following the IRC discussion, I've also replaced the macros by `static inline` functions in the header, which improves typing by the same occasion...